kimchi/src/circuits/constraints.rs

```diff
@@ -682,12 +700,33 @@ impl<F: PrimeField + SquareRootField> Builder<F> {
 
         //~ 2. Create a domain for the circuit. That is,
         //~    compute the smallest subgroup of the field that
-        //~    has order greater or equal to `n + ZK_ROWS` elements.
-        let domain_size_lower_bound =
-            std::cmp::max(gates.len(), num_lookups + 1) + ZK_ROWS as usize;
+        //~    has order greater or equal to `n + zk_rows` elements.
+        let (zk_rows, domain_size_lower_bound) = {
+            let mut zk_rows = 3;
+            let get_domain_size_lower_bound =
+                |zk_rows: u64| std::cmp::max(gates.len(), num_lookups + 1) + zk_rows as usize;
```

Comment:
It looks like `gates` might change between calls to this function so I believe it would be incorrect to compute `max` outside of the function. Mark this comment as resolved if that's indeed the case.